### PR TITLE
Issue 340: Address lodash security vulnerability

### PR DESCRIPTION
**Pull request by OldSneerJaw**
# Description

- Explicitly added lodash to the list of development dependencies and set its minimum version to 4.17.11.
- Upgraded additional development dependencies:
    - jshint: 2.10.1
    - mock-require: 3.0.3
    - nyc: 13.2.0
- Upgraded static runtime dependencies:
    - commander: 2.19.0
    - indent.js: 0.3.4
    - joi: 14.3.1

# Testing

Follow these instructions for both Sync Gateway [1.5.2](https://packages.couchbase.com/releases/couchbase-sync-gateway/1.5.2/couchbase-sync-gateway-community_1.5.2_x86_64.tar.gz) and [2.1.2](https://packages.couchbase.com/releases/couchbase-sync-gateway/2.1.2/couchbase-sync-gateway-community_2.1.2_x86_64.tar.gz) (instructions borrowed from pull request #328):

1. Generate a sync function from `conditional-validation-doc-definitions.js`:

```bash
./make-sync-function -j test/resources/conditional-validation-doc-definitions.js build/test-sync-function.js
```

2. Copy and paste the generated sync function into a Sync Gateway configuration file.
3. Attempt to create a document where neither condition is satisfied. Should be rejected with 403 Forbidden.

```
PUT /test/conditional-test-1 HTTP/1.1
Host: localhost:4985
Content-Type: application/json

{
	"type": "conditionalValidationDoc",
	"staticParentObjectProp": {
		"conditionalValidationProp": false
	}
}
```

4. Attempt to create a document where the first condition is satisfied (the value is a string) but it fails validation (the string is not a datetime). Should be rejected with 403 Forbidden.

```
PUT /test/conditional-test-1 HTTP/1.1
Host: localhost:4985
Content-Type: application/json

{
	"type": "conditionalValidationDoc",
	"staticParentObjectProp": {
		"conditionalValidationProp": "not-a-datetime"
	}
}
```

5. Attempt to create a document where the second condition is satisfied (the value is a number) but it fails validation (the number is not an integer and the number is less than the minimum value: 1533945600000). Should be rejected with 403 Forbidden.

```
PUT /test/conditional-test-1 HTTP/1.1
Host: localhost:4985
Content-Type: application/json

{
	"type": "conditionalValidationDoc",
	"staticParentObjectProp": {
		"conditionalValidationProp": 52.5
	}
}
```

6. Create a document where the first condition is satisfied (the value is a string) with a valid datetime. Should be accepted with 201 Created.

```
PUT /test/conditional-test-1 HTTP/1.1
Host: localhost:4985
Content-Type: application/json

{
	"type": "conditionalValidationDoc",
	"staticParentObjectProp": {
		"conditionalValidationProp": "2018-08-15T20:58:37-07:00"
	}
}
```

7. Attempt to replace the document with a value that is a valid integer but stills fails validation because the first condition is satisfied (i.e. it expects a datetime string but it's actually a number) and the surrounding conditional validator specifies that it is immutable. Should be rejected with 403 Forbidden.

```
PUT /test/conditional-test-1?rev=1-53a83199f53242c79bca68fd56d7fd73 HTTP/1.1
Host: localhost:4985
Content-Type: application/json

{
	"type": "conditionalValidationDoc",
	"staticParentObjectProp": {
		"conditionalValidationProp": 1533945600000
	}
}
```

# Related Issue

* GitHub issue link: #340

### CODE REVIEW

File: lib/commander/index.js [hunk 1]
```diff
@@ -580,9 +580,9 @@ Command.prototype.executeSubCommand = function(argv, args, unknown) {
   proc.on('close', process.exit.bind(process));
```
Comment (OldSneerJaw) [inline lib/commander/index.js]: Copied verbatim from https://github.com/tj/commander.js/blob/v2.19.0/index.js

File: lib/indent.js/indent.js [hunk 2]
```diff
@@ -10,6 +10,25 @@
 var indent = (function (root) {
```
Comment (OldSneerJaw) [inline lib/indent.js/indent.js]: Copied verbatim from https://github.com/zebzhao/indent.js/blob/v0.3.4/lib/indent.js

File: lib/joi/VERSION [hunk 3]
```diff
@@ -1 +1 @@
-13.7.0
+14.3.1
```
Comment (OldSneerJaw) [inline lib/joi/VERSION]: This package's update was regenerated using the instructions from https://github.com/Kashoo/synctos/blob/v2.7.0/lib/joi/README.md

**Comment by ziemek** [comment]
👍 looks good.

**Pull request merged**